cloudinit/sources/DataSourceAzure.py

```diff
@@ -1321,6 +1321,9 @@ def parse_network_config(imds_metadata):
             LOG.debug('Azure: generating network configuration from IMDS')
             network_metadata = imds_metadata['network']
             for idx, intf in enumerate(network_metadata['interface']):
+                # First IPv4 and/or IPv6 address will be obtained via DHCP.
+                # Any additional IPs of each type will be set as static
+                # addresses with a route matching the subnet prefix.
```

Comment:
Let's drop the 'with a route matching the subnet prefix'  Right? Aren't we just appending the addresses and netmask cidr value?

Reply:
We are appending the netmask CIDR prefix from subnet. Doesn't that add a route config entry as a result?

Reply:
tl;dr  we're not emitting static routes for additional IPs within a subnet (and we don't need to)

1)  the ips are per subnet, which has a fixed CIDR, so they won't add any additional route entry just for an additional IP address that's in the same subnet
2) even if we did have additional subnets outside of the primary interface subnet, the code does not emit any route specific metric values; 
3) if we needed to explicitly create a higher metric subnet route for a particular IP, then we need to emit static route configuration in netplan which we are not doing at this time

Reply:
the language suggests that we're emitting a route entry; where what actually happens is when the static ip is added, the kernel will create a subnet route by default if one does not exist; we don't control this exclusively and in particular, we're certainly not setting the metric on it.

Reply:
+1 @raharper.  Makes sense and dropping the comment